From 26abcb62d4838774956cebaa66465d56e7375904 Mon Sep 17 00:00:00 2001
From: Nathan Pratta Teodosio <nathan.teodosio@canonical.com>
Date: Thu, 29 Aug 2024 09:25:13 +0200
Subject: [PATCH 1/2] Be more verbose about permissions of /tmp/{,.X11-unix}.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3984>
(cherry picked from commit 4dbaa8178116649076bf542a5eaf42d4ce804e5b)
---
 src/wayland/meta-xwayland.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/wayland/meta-xwayland.c b/src/wayland/meta-xwayland.c
index e405161c1d..98c0d502f5 100644
--- a/src/wayland/meta-xwayland.c
+++ b/src/wayland/meta-xwayland.c
@@ -545,8 +545,10 @@ ensure_x11_unix_perms (GError **error)
   if (x11_tmp.st_uid != tmp.st_uid && x11_tmp.st_uid != getuid ())
     {
       g_set_error (error, G_IO_ERROR, G_IO_ERROR_PERMISSION_DENIED,
-                   "Wrong ownership for directory \"%s\"",
-                   X11_TMP_UNIX_DIR);
+                   "Wrong ownership for directory \"%s\", owned by %d but "
+                   "should be same as %s (owned by (%d)) or %d",
+                   X11_TMP_UNIX_DIR, x11_tmp.st_uid, TMP_UNIX_DIR, tmp.st_uid,
+                   getuid ());
       return FALSE;
     }
 
-- 
2.50.0

